mitodl/open-discussions
# Search for bootcamps

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #2043 

#### What's this PR do?
- Modifies the frontend search query code to handle bootcamps and courses.
- Adds a search facet for 'learning resource' that includes courses and bootcamps
- Adds CompactBootcampDisplay and ExpandedBootcampDisplay components to display Bootcamp results
- renames `CourseDrawer` to `LearningResourceDrawer` and modifies it to handle different object types instead of just courses.

#### How should this be manually tested?
- Run `docker-compose run web python manage.py recreate_index`
- Import some bootcamps if you haven't already done so:
```python
from course_catalog.tasks import *
get_bootcamp_data()
```
- Import some course data if you haven't already done so:
```
sync_and_upload_edx_data()
```
- Go to `/courses/search`.  There should be a new facet for `Learning Resource` that includes checkboxes for 'Course' and 'Bootcamp'.  Try these out and ensure that you can see both Course and Bootcamp records returned, and that clicking on a result opens the right-side drawer containing the expanded display for the course/bootcamp.

#### Screenshots (if appropriate)
**Learning Resource Facet**
<img width="1184" alt="Screen Shot 2019-07-23 at 2 18 53 PM" src="https://user-images.githubusercontent.com/187676/61739718-03451900-ad5b-11e9-8837-06037de5cfa8.png">

**Learning Resource Facet, mobile**
<img width="411" alt="Screen Shot 2019-07-23 at 2 19 35 PM" src="https://user-images.githubusercontent.com/187676/61739951-78b0e980-ad5b-11e9-855d-2235ead264c4.png">


**Bootcamp facet selected, bootcamp card displayed**
<img width="1185" alt="Screen Shot 2019-07-23 at 2 19 06 PM" src="https://user-images.githubusercontent.com/187676/61739864-4e5f2c00-ad5b-11e9-9214-fe2204ef48a4.png">

** Bootcamp expanded display **
<img width="928" alt="Screen Shot 2019-07-23 at 2 32 20 PM" src="https://user-images.githubusercontent.com/187676/61739914-69ca3700-ad5b-11e9-84fb-2e9ee225d05c.png">
